src/main/java/net/revelc/code/formatter/json/JsonFormatter.java

```diff
@@ -79,11 +79,10 @@ public DefaultPrettyPrinter withSeparators(final Separators separators) {
     protected String doFormat(final String code, final LineEnding ending) throws IOException {
         try (StringWriter stringWriter = new StringWriter()) {
             JsonParser jsonParser = this.formatter.createParser(code);
-            // note: line ending set in init for this usecase
             final Iterator<Object> jsonObjectIterator = this.formatter.readValues(jsonParser, Object.class);
             while (jsonObjectIterator.hasNext()) {
                 String jsonString = this.formatter.writer().writeValueAsString(jsonObjectIterator.next());
-                stringWriter.write(jsonString);
+                stringWriter.write(jsonString.strip().replaceAll("\\R", ending.getChars()));
```

Comment:
Probably leaving a line comment that 

> `\R` is a special escape for regex engine meaning "line break"

will be a plus.

Reply:
In general, I don't feel the need to fill code with comments that programmers can easily look up, and Java's regex escape patterns are pretty well known and easily looked up (https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html).

But, I found a better solution anyway... I pre-compiled the pattern and gave it a useful name. I think that satisfies your comment and also makes the code more efficient.